### ISSUE TITLE
variable-name rule ignores interface properties

### ISSUE THREAD

**Issue by Calidus** [palantir/tslint#4691]
### Bug Report

- __TSLint version__: 5.12.1
- __TypeScript version__: 3.1.6
- __Running TSLint via__:  VSCode 

#### TypeScript code being linted

```ts
export interface Foo {
    Bar: number
    string: string;
}
```

with `tslint.json` configuration:

```json
"variable-name": [true, "ban-keywords", "check-format"],
```

#### Actual behavior
No Errors
#### Expected behavior
Two linting errors, one for a banned key word string and the other for not using camelCase.

**Comment by JoshuaKGoldberg** [comment]
Yes, that's the intended behavior. The *variable*-name rule only checks variables.

Whether it is desirable to add a separate rule for property names on interfaces and/or type literals is a separate, interesting question. Is this a common enough use case to justify adding into TSLint now, given that the library is being deprecated? #4534 

There are cases where interfaces need to describe external shapes (e.g. static prototypes and third party shenanigans). If the rule is to go in, there should be a proposal for how to deal with those.

**Comment by Calidus** [comment]
I was surprised that variable name applied to public class fields but not interface members. Then it gets even more awkward if you implement a interface that does not following the defined rules for class fields.

```
export interface Foo {
    Bar: number;
    string: string;
}
export class MyClass implements Foo {
    public Bar: number; //variable-name warning
    public string: string; //variable-name warning
}
```

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!
